README.md

```diff
@@ -291,6 +296,125 @@ class YourClass {
+
+```java
+ExemplarSampler myExemplarSampler = new MyExemplarSampler();
+ExemplarConfig.setCounterExemplarSampler(myExemplarSampler);
+ExemplarConfig.setHistogramExemplarSampler(myExemplarSampler);
+```
+
+Note that `ExemplarConfig` is used when a metric is created, so changing the default has no effect on metrics that have already been created.
+
+### Per Metric Exemplar Samplers
+
+The metric builders for `Counter` and `Histogram` have methods for setting the exemplar sampler for that individual metric. This takes precedence over the global setting in `ExemplarConfig`.
+
+The following calls enable the default exemplar sampler for individual metrics. This is useful if you disabled the exemplar sampler globally with `ExemplarConfig.disableExemplarSamplers()`.
+
+```java
+Counter myCounter = Counter.build()
+    .name("number_of_events_total")
+    .help("help")
+    .withExemplarSampler()
+    ...
+    .register();
+```
+
+```java
+Histogram myHistogram = Histogram.build()
+    .name("my_latency")
+    .help("help")
+    .withExemplarSampler()
+    ...
+    .register();
+```
+
+The following calls disable the default exemplar sampler for individual metrics:
+
+```java
+Counter myCounter = Counter.build()
+    .name("number_of_events_total")
+    .help("help")
+    .withoutExemplarSampler()
+    ...
+    .register();
+```
+
+```java
+Histogram myHistogram = Histogram.build()
+    .name("my_latency")
+    .help("help")
+    .withoutExemplarSampler()
+    ...
+    .register();
+```
+
+The following calls set a custom `MyExemplarSampler` for individual metrics:
+
+```java
+// CounterExemplarSampler
+Counter myCounter = Counter.build()
+    .name("number_of_events_total")
+    .help("help")
+    .withExemplarSampler(new MyExemplarSampler())
+    ...
+    .register();
+```
+
+```java
+// HistogramExemplarSampler
+Histogram myHistogram = Histogram.build()
+    .name("my_latency")
+    .help("help")
+    .withExemplarSampler(new MyExemplarSampler())
+    ...
+    .register();
+```
+
+### Per Observation Exemplars
+
+You can explicitly create an exemplar for an individual observation. This takes precedence over the exemplar sampler configured with the metric.
+
+The following call will increment a counter, and create an exemplar with the specified `span_id` and `trace_id` labels:
+
+```java
+myCounter.incWithExemplar(Exemplar.SPAN_ID, "abcdef", Exemplar.TRACE_ID, "123456");
+```
+
+The following call will observe a value of `0.12` in a histogram, and create an exemplar with the specified `span_id` and `trace_id` labels:
+
+```java
+myHistogram.observeWithExemplar(0.12, Exemplar.SPAN_ID, "abcdef", Exemplar.TRACE_ID, "123456");
+```
+
+All methods for observing and incrementing values have `...withExemplar` equivalents. There are versions taking the exemplar labels as a `String...` as shown in the example, as well as versions taking the exemplar labels as a `Map<String, String>`.
```

Comment:
Have you thought about doing this as part of inc() call? This isn't Go, function overloading is possible and normal in Java.

Reply:
If someone wanted an explicitly empty exemplar, they could pass an empty map.